server/trafficrules/src/main/java/com/coffeejug/trafficrules/web/UserController.java

```diff
@@ -0,0 +1,54 @@
+package com.coffeejug.trafficrules.web;
+
+import com.coffeejug.trafficrules.dto.UserCodeDto;
+import com.coffeejug.trafficrules.projection.ProgressPrj;
+import com.coffeejug.trafficrules.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+
+    @GetMapping("/v1/user/code")
+    public ResponseEntity<UserCodeDto> generateUserCode() {
+        return new ResponseEntity<UserCodeDto>(userService.generateUserCode(), HttpStatus.OK);
+    }
+
+    @GetMapping("/v1/user/progress")
+    public ResponseEntity<List<ProgressPrj>> getUserProgress(@RequestParam String code) {
+
+        if (code == null || code.length() != 10) {
+            return new ResponseEntity("Wrong code", HttpStatus.BAD_REQUEST);
+        }
+        if (!userService.existsByCode(code)) {
+            return new ResponseEntity("Code does not exists", HttpStatus.BAD_REQUEST);
+        }
+        return new ResponseEntity<>(userService.getUserProgress(code), HttpStatus.OK);
+    }
+
+    @PostMapping("/v1/user/progress")
+    private ResponseEntity saveProgress(@RequestParam String code,

```

Comment:
We can just return the needed entity without wrapping it to `ResponseEntity`

Reply:
What do you mean ?

Reply:
In this case, the method could be `void` in other cases it can directly return `UserCodeDto` or any other type that is needed

Reply:
But how frontend will know, that everything is OK (200) ?

Reply:
Spring will return 200 Ok. It can abstract `ResponseEntity` for us